los-deferal-02/Hello-Books
# #166237755 Setup Automated Testing Framework

#### What does this PR do?
Set up the testing framework for the project

#### Description of Task to be completed?
- use combination of mocha, chai, and chai-http as testing framework
- add test script to package.json file
- write test files for root and undefined routes

#### How should this be manually tested?
On Terminal:
- Run `git clone https://github.com/los-deferal-02/Hello-Books.git` to have this repo on your local machine
- Navigate to the repo directory by running `cd Hello-Books/`
- Ensure your working directory is clean by running  `git status`  command
- Get a copy of the PR by typing  `git fetch origin pull/6/head:ch-setup-testing-framework-166237755`
- Now that you have a copy of the branch, switch to it using  `git checkout ch-setup-testing-framework-166237755`
- Your directory will now be an exact copy of the PR
- Run `npm install` to install all project's dependencies
- Type `npm test` to run the automated tests

#### Any background context you want to provide?
To ensure the back end has all functionality working as intended, it is recommended to have testing framework set up to serve as a watchdog for unintended errors.

#### What are the relevant pivotal tracker stories?
[#166237755](https://www.pivotaltracker.com/story/show/166237755)

#### Screenshots (if appropriate)
![setup-testing-framework-scrnshot](https://user-images.githubusercontent.com/32547932/58517775-81929e00-81a4-11e9-9b7d-eb2b6eed3af0.png)
